src/SBInclinedExponential.cpp

```diff
@@ -0,0 +1,286 @@
+    }
+
+    // NB.  This function is virtually wrapped by repr() in SBProfile.cpp
+    std::string SBInclinedExponential::SBInclinedExponentialImpl::serialize() const
+    {
+        std::ostringstream oss(" ");
+        // NB. The choice of digits10 + 4 is because the normal general output
+        // scheme for double uses fixed notation if >= 0.0001, but then switches
+        // to scientific for smaller numbers.  So those first 4 digits in 0.0001 don't
+        // count for the number of required digits, which is nominally given by digits10.
+        // cf. http://stackoverflow.com/questions/4738768/printing-double-without-losing-precision
+        // Unfortunately, there doesn't seem to be an easy equivalent of python's %r for
+        // printing the repr of a double that always works and doesn't print more digits than
+        // it needs.  This is the reason why we reimplement the __repr__ methods in python
+        // for all the SB classes except SBProfile.  Only the last one can't be done properly
+        // in python, so it will use the C++ virtual function to get the right thing for
+        // any subclass.  But possibly with ugly extra digits.
+        oss.precision(std::numeric_limits<double>::digits10 + 4);
+        oss << "galsim._galsim.SBInclinedExponential("<<getInclination()<<", "<<getScaleRadius()<<", "<<getScaleHeight();
+        oss <<", "<<getFlux()<<", False";
+        oss << ", galsim.GSParams("<<*gsparams<<"))";
+        return oss.str();
+    }
+
+    SBInclinedExponential::SBInclinedExponentialImpl::SBInclinedExponentialImpl(Angle inclination, double scale_radius,
+            double scale_height, double flux, const GSParamsPtr& gsparams) :
+        SBProfileImpl(gsparams),
+        _inclination(inclination),
+        _r0(scale_radius),
+        _h0(scale_height),
+        _flux(flux),
+        _inv_r0(1./_r0)
+    {
+        dbg<<"Start SBInclinedExponential constructor:\n";
+        dbg<<"inclination = "<<_inclination<<std::endl;
+        dbg<<"scale radius = "<<_r0<<std::endl;
+        dbg<<"scale height = "<<_h0<<std::endl;
+        dbg<<"flux = "<<_flux<<std::endl;
+
+        // Check if cos(inclination) is within allowed limits, and institute special handling if it isn't
+
+        double cosi = std::abs(_inclination.cos());
+
+        if(cosi<sbp::minimum_cosi)
+        {
+            // Perfectly edge-on isn't analytic, so we truncate at the minimum cos(inclination) value
+            cosi = sbp::minimum_cosi;
+        }
+
+        // Now set up, using this value of cosi
+
+        _r0_cosi = _r0*cosi;
+        _inv_r0_cosi = 1./_r0_cosi;
+
+        _h_tani_over_r = scale_height*std::abs(inclination.sin())*_inv_r0_cosi; // A tiny bit more accurate than using tan of
+                                                                                // the truncated value
+
+        _half_pi_h_tani_over_r = 0.5*M_PI*_h_tani_over_r;
```

Comment:
The reason I saw this as problematic wasn't due to the convolution factor. Rather, it has to do with the inclination of the exponential disk. See the maths here for the basis for the calculation here: https://www.dropbox.com/s/z6h14bgd199czsi/Inclined_Exponential.pdf?dl=0 - in particular, look at equation (3), which is before any coordinate transformation. The issue comes from the y^2/cos^2(i) term in the exponential on the right side. In the case where i=pi/2, the assumption that we can integrate across exponential disks breaks down, so instead we have to integrate through a disk, and this doesn't have a closed-form solution that I know of.

The possible problem in Fourier space comes from the fact that as i approaches pi/2, ky*R*cos(i) approaches zero, but the limits for ky approach infinity, resulting in a 0/0 type situation. However, I think the convolution saves us here. Since we multiply by ky/sinh(ky), the k value goes to zero at a non-infinite value. This goes back to your other comment that my calculation of the maximum k was too conservative, since I didn't account for this. So, I think once that issue is resolved, the edge-on case won't be an issue either (it will just take a bit of rewriting the code). I'll have a look at this tomorrow.